abstract/base-test.js

```diff
@@ -0,0 +1,54 @@
+module.exports = function (test, level, location) {
+  test('Level constructor provides access to levelup errors', function (t) {
+    t.ok(level.errors, '.errors property set on constructor')
+    t.end()
+  })
+
+  test('test db open and use, level(location, cb)', function (t) {
+    level(location, function (err, db) {
+      t.notOk(err, 'no error')
+      db.put('test1', 'success', function (err) {
+        t.notOk(err, 'no error')
+        db.close(t.end.bind(t))
+      })
+    })
+  })
+
+  test('test db open and use, level(location, options, cb)', function (t) {
+    level(location, { createIfMissing: false, errorIfExists: false }, function (err, db) {
```

Comment:
Maybe this should be in a separate test file too, because it uses `createIfMissing` and `errorIfExists` options?

Reply:
Gah, these tests are connected. `db-values-test.js` will error if `nonPersistent` is `undefined` and `test2` key hasn't been set.

Reply:
What if `db-values-test.js` always made a `db.batch()` and with it's own keys? We skip the `nonPersistent` parameter.

Reply:
Actually, I'd prefer if we could leave this as is for now.